src/installer/tests/TestUtils/FileUtils.cs

```diff
@@ -28,5 +30,34 @@ public static void DeleteFileIfPossible(string filePath)
             {
             }
         }
+
+        /// <summary>
+        /// Writes JSON content to a file with optional UTF8 BOM.
+        /// </summary>
+        /// <param name="filePath">The path to the file to write</param>
+        /// <param name="jsonContent">The JSON content to write</param>
+        /// <param name="withUtf8Bom">Whether to include UTF8 BOM (0xEF, 0xBB, 0xBF) at the beginning</param>
+        public static void WriteJsonWithOptionalUtf8Bom(string filePath, string jsonContent, bool withUtf8Bom)
```

Comment:
Maybe `WriteAllTextWithUtf8Bom` and move the `withUtf8Bom` condition to be handled in `RuntimeConfig.Save`